docs/DeveloperGuide.md

```diff
@@ -5,9 +5,146 @@
+
+- Step 3: The obtained information is sent to the private method `ParserCalories#makeNewInputEntry(String, int, String)` to create a new entry of class `InputEntry` that extends `Entry`. An `Entry` object is then returned to the caller, `CalorieList#addEntry(String)`.
+
+- Step 4: The returned `Entry` object is added into the `calorieArrayList` member of type `ArrayList<Entry>` in the `CalorieList`, via the `ArrayList.add()` method.
+
+- Step 5: `FileHandler#updateFile()` is then called to update the data file with the new entry in the `CalorieList`.
+
+The sequence diagram for this feature is shown below:
+
+![CaloriesAddEntrySeqDiagram](CaloriesAddEntrySeqDiagram.png)
+
+### Calculating calorie requirements based on a user`s goals
+
+#### Implementation
+
+This functionality is facilitated by `UserGoals`. It implements one operation, namely:
+- `UserGoals#getHealthInfo(User)`
+
+This operation is exposed in the `User` class as `User#getHealhInfo()`.
+
+Given below is an example usage scenario and how this mechanism behaves at every step:
+- Step 1: When the user inputs the command `user progress` in the terminal,
+  the string is sent to `User#getHealthInfo()`, which calls `UserGoals#getHealthInfo(User)`.
+
+- Step 2: The method retrieves the user`s information such as his height, weight, age, gender, exercise levels and intended goal.
+
+- Step 3: Using these information, the method creates a `requestBody` `String`. 
+
+- Step 4: The created `requestBody` is used to send a `HttpRequest` to RapidAPI`s Fitness API, and the response is parsed to determine the number of calories a user needs to consume according to their personal goals.
+
+- Step 5: This value is set to `User.caloriesRequired` by `User#setCaloriesRequired(int)`.
+
+#### Design considerations
+
+- **Alternative 1 (current choice):** Uses an API to get the calories needed
+    - Pros: No need to figure out the optimal algorithm
+    - Cons: Need to parse response to sieve out necessary information
+
+- **Alternative 2:** Uses an algorithm to find the number of calories needed
+  - Pros: Not dependent on external APIs
+  - Cons: Need to come up with an algorithm to use
+
+### Calories list feature
+
+The `calories list` feature lists out the record of all the Calories data that the user has keyed in. The Calories data are all stored into a `ArrayList<Entry> calorieArrayList` attribute of the `CalorieList` Class. Calories data are printed when the `printCalorieList()` function is called. 
+
+The `printCalorieList()` function iterates through the `calorieArrayList` and prints out the Entries according to its order in the Array List.
+
+The Class diagram for Calories list feature is shown below. Unrelated attributes and Classes were excluded.
+
+![CaloriesListClassDiagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/rexyyong/tp/DevGuideRex/docs/CaloriesListClassDiagram.puml)
```

Comment:
Could use these proper symbols for representing public/private classes instead of squares
+ : public
- : private
# : protected
~ : package private